csrc/ir_nodes.cpp

```diff
@@ -817,44 +816,48 @@ SqueezeOp::SqueezeOp(
 
   addOutput(out);
   addInput(in);
-  addAttribute(IrBuilder::create<Attribute<std::vector<bool>>>(
-      passkey.ir_container_, std::move(is_squeeze_dims)));
 
-  if (!out->isA<TensorView>() || !in->isA<TensorView>()) {
-    return;
-  }
-
-  auto in_tv = in->as<TensorView>();
-  auto out_tv = out->as<TensorView>();
-  auto in_dom = TensorDomain::noReductions(in_tv->getMaybeRFactorDomain());
-  auto& out_dom = out_tv->getRootDomain();
-  TORCH_INTERNAL_ASSERT(
-      is_squeeze_dims.size() == in_dom.size(),
-      "The dimensions of input tensor and does not match with is_squeeze_dims");
-
-  auto in_size = is_squeeze_dims.size();
-  auto num_removed_broadcasts = 0;
-  for (const auto i : c10::irange(is_squeeze_dims.size())) {
-    if (is_squeeze_dims[i]) {
-      num_removed_broadcasts++;
-      auto id = in_dom[i];
-      TORCH_INTERNAL_ASSERT(
-          id->isBroadcast(), "Can not squeeze non-broadcasting dimension(s).");
-      TORCH_INTERNAL_ASSERT(
-          !id->hasExpandedExtent(), "Can not squeeze expanded dimension(s).");
-      TORCH_INTERNAL_ASSERT(
-          id->extent()->isOneInt(),
-          "Can not squeeze dimension(s) with size != 1.");
-    } else {
-      auto in_id = in_dom[i];
-      auto out_id = out_dom[i - num_removed_broadcasts];
-      TORCH_INTERNAL_ASSERT(
-          in_id->sameAs(out_id), "IterDomain does not match in BroadcastOp");
+  // Validate the squeeze flags when this expr is created with
+  // TensorView. Broadcast with TensorIndex only appears after
```

Comment:
I think we will never have squeeze with tensor index. Squeeze is lowered into set op.

Reply:
That's correct. I'll change it to an assertion